contracts/MultiSend.sol

```diff
@@ -1,74 +1,117 @@
+// Copyright (C) 2018 Alon Bukai This program is free software: you 
+// can redistribute it and/or modify it under the terms of the GNU General 
+// Public License as published by the Free Software Foundation, version. 
+// This program is distributed in the hope that it will be useful, 
+// but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+// or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+// more details. You should have received a copy of the GNU General Public
+// License along with this program. If not, see http://www.gnu.org/licenses/
+
 pragma solidity ^0.4.18;
 
 import "giveth-common-contracts/contracts/Escapable.sol";
 
-// TightlyPacked is cheaper if you need to store input data and if amount is less than 12 bytes.
-// Normal is cheaper if you don't need to store input data or if amounts are greater than 12 bytes.
-// Supports deterministic deployment. As explained here: https://github.com/ethereum/EIPs/issues/777#issuecomment-356103528
+/// @notice MultiSend is a contract for sending multiple ETH/ERC20 Tokens to multiple addresses
+/// In addition this contract can call multiple contracts with multiple amounts
+/// There are also TightlyPacked functions which in some situations allow for gas savings
```

Comment:
I would include the parts that specify when each approach is cheaper, that's important information. Also, I would use ` to make `MultiSend` look all code-y